### PR TITLE
feat: Support `forwardAuthToken` for all types of destinations

### PR DESCRIPTION
**Pull request by marikaner**
This allows to set `forwardAuthToken` on all types of destinations. Previously this was only possible for registered destinations.

Closes https://github.com/SAP/cloud-sdk-backlog/issues/1121.

### CODE REVIEW

**Review by tomfrenken — APPROVED** [review]
A few minor comments, overall LGTM

File: packages/connectivity/src/scp-cf/destination/destination-from-service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      destinationName: 'FORWARD',
+      jwt: providerServiceToken
+    });
+    expect(retrievedDestination?.forwardAuthToken).toBe(true);
+    expect(retrievedDestination?.authTokens?.[0].value).toEqual(
+      providerServiceToken
+    );
+  });
+
+  it('does not execute additional auth flows, if `forwardAuthToken` is set (e.g. OAuth2ClientCredentials)', async () => {
```
[the remark this answers is not in the document]
Comment (marikaner) [reply packages/connectivity/src/scp-cf/destination/destination-from-service.spec.ts:47]: I found it difficult to make clear in the expectation what this is actually testing. Maybe in fact I should mock away the single destination call and check that this is not called.

File: packages/connectivity/src/scp-cf/destination/destination-from-vcap.spec.ts [hunk 2]
```diff
@@ -193,6 +207,66 @@ describe('vcap-service-destination', () => {
...
+      destinationName: 'my-custom-service',
+      useCache: true,
+      jwt
+    });
+
+    expect(destination?.authTokens?.[0]).toMatchObject({ value: jwt });
+  });
+
+  // TODO: fix in #1123
+  it.skip('does not cache the forwarded token', async () => {
```
[the remark this answers is not in the document]
Comment (marikaner) [reply packages/connectivity/src/scp-cf/destination/destination-from-vcap.spec.ts:248]: In this test I first call `getDestinationFromServiceBinding()` with `useCache` set to `true`. Therefore the destination is cached. I expect that the destination using `forwardAuthToken` has the token when retrieving it (see tests above). But the destination in the cache should not have the token. Currently we pass the destinations from the cache by reference, which causes it to reside in the cache anyways.

File: packages/connectivity/src/scp-cf/destination/destination-from-vcap.ts [hunk 3]
```diff
@@ -109,31 +201,23 @@ async function transform(
...
   try {
-    const jwtFromOptions = iss ? { iss } : jwt ? decodeJwt(jwt) : undefined;
-    const useCacheIgnoringUndefined =
-      typeof useCache !== 'undefined' ? { useCache } : {};
-    const destination = await destinationForServiceBinding(destinationName, {
-      serviceBindingTransformFn,
-      jwt: jwtFromOptions,
-      ...useCacheIgnoringUndefined
-    });
+    // TODO: Why did we do this?
```
[the remark this answers is not in the document]
Comment (marikaner) [reply packages/connectivity/src/scp-cf/destination/destination-from-vcap.ts:216]: yeah, not this needs to go ;) good catch 😊

**Pull request merged**